core/src/main/java/org/elasticsearch/index/shard/SearchOperationListener.java

```diff
@@ -225,5 +233,25 @@ public void onFreeScrollContext(SearchContext context) {
                 }
             }
         }
+
+        @Override
+        public void validateSearchContext(SearchContext context) throws Exception {
```

Comment:
I am not sure we should make it throws Exception, I think stuff like `IllegalArgumentException` or so would be appropriate here... maybe just don't add a checked exception?